scripts/generate_records/generate_db_records.js

```diff
@@ -0,0 +1,396 @@
+  const providerCumulusId = await loadProvider(knex);
+
+  const models = {
+    geModel: new GranulesExecutionsPgModel(),
+    executionModel: new ExecutionPgModel(),
+    granuleModel: new GranulePgModel(),
+    fileModel: new FilePgModel(),
+  };
+  const iterableParamGenerator = getBatchParamGenerator({
+    knex,
+    numberOfGranules,
+    collectionCumulusId,
+    providerCumulusId,
+    filesPerGranule,
+    granulesPerBatch,
+    executionsPerBatch,
+    models,
+    variance,
+    swallowErrors,
+  });
+  await pMap(
+    iterableParamGenerator,
+    // this lambda function swallows uploadDataBatch's return to prevent ballooning memory use
+    async (params) => {
+      await uploadDataBatch(params);
+    },
+    { concurrency }
+  );
+};
+
+/**
+ * Parse executions per batch and granules per batch based on a given ratio
+ *
+ * @param {string} executionsPerGranule - executionsPerGranule in <executions>:<granules>
+ * @returns {{executionsPerBatch: number, granulesPerBatch: number}}
+ */
+const parseExecutionsGranulesBatch = (executionsPerGranule) => {
+  // expect to come in format 'x:y'
+  try {
+    const split = executionsPerGranule.split(':');
+    if (split.length < 2) {
+      throw new Error(`only 1 value could be split from ${executionsPerGranule}`);
+    }
+    const executionsPerBatch = Number.parseInt(split[0], 10);
+    const granulesPerBatch = Number.parseInt(split[1], 10);
+    return { executionsPerBatch, granulesPerBatch };
+  } catch (error) {
+    throw new Error(`cannot parse ${executionsPerGranule}, expected format <executions>:<granules> ratio \n${error}`);
+  }
+};
+
+/**
+ * parse command line args for run parameters
+ *
+ * @returns {{
+ *   granules: number,
+ *   files: number,
+ *   granulesPerBatch: number
+ *   executionsPerBatch: number
+ *   collections: number
+ *   concurrency: number
+ *   variance: boolean
+ *   swallowErrors: boolean
+ * }}
+ */
+const parseArgs = () => {
+  const {
+    granulesK,
+    files,
+    executionsPerGranule,
+    collections,
+    variance,
+    concurrency,
+    swallowErrors,
+  } = minimist(
+    process.argv,
+    {
+      string: [
+        'collections',
+        'files',
+        'granulesK',
+        'executionsPerGranule',
+        'concurrency',
+      ],
+      boolean: [
+        'swallowErrors',
+        'variance',
+      ],
+      alias: {
+        g: 'granulesK',
+        f: 'files',
+        c: 'collections',
+        e: 'executionsPerGranule',
+        C: 'concurrency',
+        v: 'variance',
+        s: 'swallowErrors',
+      },
+      default: {
+        collections: process.env.COLLECTIONS || 1,
+        files: process.env.FILES || 1,
+        granulesK: process.env.GRANULES_K || 10,
+        executionsPerGranule: process.env.EXECUTIONS_PER_GRANULE || '2:2',
+        variance: process.env.VARIANCE || false,
+        concurrency: process.env.CONCURRENCY || 1,
+        swallowErrors: process.env.SWALLOW_ERRORS || true,
```

Comment:
I don't think you need to change anything, but just wanted to note that I feel a bit odd about the default for swallowErrors is true if nothing is passed in as an option, and it is set that way here, _yet_ everywhere else, swallowErrors is default set to false.

Reply:
I've solved this by rethinking how this works, it's not set as a cmd line arg but just using pMap's stopOnError arg. (means whole batches can be lost if an error happens, but we ended up doing really small batches for high concurrency in practice so a potential loss like that is a drop in the bucket
https://github.com/nasa/cumulus/pull/3679/commits/285a0733acfb78aeb9f59bdf016298739686ef8e